superset/datasets/dao.py

```diff
@@ -0,0 +1,119 @@
+from sqlalchemy.exc import SQLAlchemyError
+
+from superset import db
+from superset.connectors.sqla.models import SqlaTable
+from superset.models.core import Database
+from superset.views.base import DatasourceFilter
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetDAO:
+    @staticmethod
+    def get_owner_by_id(owner_id: int) -> Optional[object]:
+        return (
+            db.session.query(current_app.appbuilder.sm.user_model)
+            .filter_by(id=owner_id)
+            .one_or_none()
+        )
+
+    @staticmethod
+    def get_database_by_id(database_id) -> Optional[Database]:
+        try:
+            return db.session.query(Database).filter_by(id=database_id).one_or_none()
+        except SQLAlchemyError as e:  # pragma: no cover
+            logger.error(f"Could not get database by id: {e}")
+            return None
+
+    @staticmethod
+    def validate_table_exists(database: Database, table_name: str, schema: str) -> bool:
+        try:
+            database.get_table(table_name, schema=schema)
+            return True
+        except SQLAlchemyError as e:  # pragma: no cover
+            logger.error(f"Got an error {e} validating table: {table_name}")
+            return False
+
+    @staticmethod
+    def validate_uniqueness(database_id: int, name: str) -> bool:
+        dataset_query = db.session.query(SqlaTable).filter(
+            SqlaTable.table_name == name, SqlaTable.database_id == database_id
+        )
+        return not db.session.query(dataset_query.exists()).scalar()
+
+    @staticmethod
+    def validate_update_uniqueness(
+        database_id: int, dataset_id: int, name: str
+    ) -> bool:
+        dataset_query = db.session.query(SqlaTable).filter(
+            SqlaTable.table_name == name,
+            SqlaTable.database_id == database_id,
+            SqlaTable.id != dataset_id,
+        )
+        return not db.session.query(dataset_query.exists()).scalar()
+
+    @staticmethod
+    def find_by_id(model_id: int) -> SqlaTable:
+        data_model = SQLAInterface(SqlaTable, db.session)
+        query = db.session.query(SqlaTable)
+        query = DatasourceFilter("id", data_model).apply(query, None)
+        return query.filter_by(id=model_id).one_or_none()
+
+    @staticmethod
+    def create(properties: Dict) -> Optional[SqlaTable]:
+        model = SqlaTable()
+        for key, value in properties.items():
+            setattr(model, key, value)
+        try:
+            db.session.add(model)
+            db.session.commit()
+        except SQLAlchemyError as e:  # pragma: no cover
+            logger.error(f"Failed to create dataset: {e}")
+            db.session.rollback()
+            return None
+        return model
+
+    @staticmethod
+    def update(model: SqlaTable, properties: Dict) -> Optional[SqlaTable]:
+        for key, value in properties.items():
+            setattr(model, key, value)
+        try:
+            db.session.merge(model)
+            db.session.commit()
```

Comment:
Not sure we want TX to finish in these methods. What happens when you want to do a few things as part of a single atomic unit of work? (I.e. update a few items via a DAO, and one fails)

Reply:
how about making an optional parameter for this? We can default to commit and on the cases (probably a minority) where we need complex transactions call it like `create(data, commit=False)` ?

Reply:
That works. I would also try and solve it as an aspect, maybe with a quick decorator? Don't want to push this off too much - we can always add the decorator later...

Reply:
Not getting the idea of a decorator here, to switch commit's on and off?
I propose we talk about it and push it to the next PR for dataset's columns